Use SPDX license expression

cdce8p opened the pull request:
<!-- Thank you for your contribution! -->

## What do these changes do?
SSIA

<!-- Please give a short brief about these changes. -->

## Are there changes in behavior for the user?

<!-- Outline any notable behaviour for the end users. -->

## Related issue number

<!-- Are there any issues opened that will be resolved by merging this change? -->

## Checklist

- [ ] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] If you provide code modifications, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep the list in alphabetical order, the file is sorted by name.
- [ ] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

webknjaz commented on CHANGES/639.packaging.rst:
```diff
@@ -0,0 +1 @@
+Use SPDX license identifier for project metadata.
```
Please change the writing style from a "TODO-list item" to "X changed" or "X is now Y".

cdce8p commented:
> Additionally, PEP 639 deprecates the trove classifiers, so their removal should be coupled with SPDX.

Updated the changelog and removed the classifier. Do note though that setuptools will continue to use the `License` field in the project metadata until the metadata is moved to `pyproject.toml`. If you like, I can do that in a followup.

webknjaz commented:
@cdce8p I'm not yet ready to switch to PEP 621. But we could give it a try and see..
